### PR TITLE
feat: add Slice and Custom Type asset management hooks

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR implements 6 new hooks to facilitate general Slice and Custom Type asset management:

- `slice:asset:update`
- `slice:asset:read`
- `slice:asset:delete`
- `custom-type:asset:update`
- `custom-type:asset:read`
- `custom-type:asset:delete`

All `*:asset:*` hooks are **required** by adapters as they implement crucial functionality needed by Slice Machine.

These hooks would be used for the following purposes

- Saving screenshots
- Saving mock values and/or configuration
- Saving any Slice- or Custom Type-specific asset-like data

For details on the technical decisions made behind these hooks, see the following comment thread: https://github.com/prismicio/slicemachine-plugin-kit/pull/2#pullrequestreview-1150680708

### Refactors

- Move all hook types from `src/types.ts` into per-hook files in `src/hooks`. The number of hooks and the verbosity of the types outgrew the single `types.ts` file.
- Add TSDocs to all hook types.

## Checklist:

- [ ] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

🦒

### CODE REVIEW

**Review by lihbr — APPROVED** [review]
LGTM so far (not sure if it's still draft)

**Comment by angeloashmore** [comment]
I need to do one final check on my side before merging, but it should be all good.

If you want to move forward and build on top of this PR, feel free to merge. 🙂

**Pull request merged**
